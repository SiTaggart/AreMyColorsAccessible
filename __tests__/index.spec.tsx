/* eslint-env jest */
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import HomePage from '../pages';

describe('Home Page', (): void => {
  it('should render without crashing', (): void => {
    ReactDOM.render(
      // @ts-ignore
      <HomePage />,
      document.createElement('div')
    );

    const homeCmp = renderer
      .create(
        // @ts-ignore
        <HomePage />
      )
      .toJSON();
    expect(homeCmp).toMatchSnapshot();
  });

  it('should render based on query string props', (): void => {
    const homeCmp = renderer
      .create(
        <HomePage
          query={{
            background: '#5E38A8',
            isLight: false,
            textColor: '#eee',
            colorCombos: []
          }}
        />
      )
      .toJSON();
    expect(homeCmp).toMatchSnapshot();
  });
});
