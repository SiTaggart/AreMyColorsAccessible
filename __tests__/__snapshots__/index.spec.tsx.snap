// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render based on query string props 1`] = `
@media (min-width: 768px) {
  .emotion-0 {
    height: 100%;
  }
}

.emotion-2 {
  position: relative;
}

@media (min-width: 768px) {
  .emotion-2 {
    min-height: 85%;
  }
}

.emotion-4 {
  margin: 0 auto;
  max-width: 1150px;
  padding: 0 1rem;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 2rem;
  text-align: center;
}

@media (min-width: 768px) {
  .emotion-6 {
    margin-bottom: 3rem;
  }
}

@media (min-width: 1024px) {
  .emotion-6 {
    margin-bottom: 5rem;
  }
}

.emotion-8 {
  font-weight: 700;
  font-size: 20vw;
  margin: 3rem 0;
  width: 100%;
}

@media (min-width: 768px) {
  .emotion-8 {
    font-size: 18vw;
    margin: 4rem 0;
  }
}

@media (min-width: 1024px) {
  .emotion-8 {
    font-size: 15vw;
    margin: 5rem 0;
  }
}

@media (min-width: 1480px) {
  .emotion-8 {
    font-size: 9vw;
  }
}

.emotion-10 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-bottom: 2rem;
  width: 100%;
}

@media (min-width: 480px) {
  .emotion-10 {
    width: 50%;
  }
}

@media (min-width: 1024px) {
  .emotion-10 {
    width: 25%;
  }
}

.emotion-12 {
  font-size: 14pt;
  margin: 0;
}

.emotion-14 {
  font-weight: 700;
  font-size: 10vw;
  margin: 0;
}

@media (min-width: 768px) {
  .emotion-14 {
    font-size: 6vw;
  }
}

@media (min-width: 1024px) {
  .emotion-14 {
    font-size: 4vw;
  }
}

@media (min-width: 1480px) {
  .emotion-14 {
    font-size: 3vw;
  }
}

.emotion-24 {
  font-size: 18pt;
  margin: 0;
}

.emotion-34 {
  margin: 0 auto;
  max-width: 1200px;
}

.emotion-36 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 auto;
  max-width: 1200px;
  padding: 0 1rem;
  -webkit-transition: color 400ms ease-out;
  transition: color 400ms ease-out;
}

.emotion-38 {
  padding: 1rem;
  width: 100%;
}

@media (min-width: 768px) {
  .emotion-38 {
    -webkit-flex: 1 1 auto;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
    width: 50%;
  }
}

.emotion-40 {
  display: inline-block;
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-42 {
  background: rgba(255, 255, 255, 0.1);
  border: solid 1px;
  border-radius: 3px;
  font-size: 1.2rem;
  padding: 0.5rem 1rem;
  margin-bottom: 2rem;
  width: 100%;
}

.emotion-44 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (min-width: 1024px) {
  .emotion-44 {
    margin: 0 -0.5rem;
  }
}

.emotion-46 {
  width: 100%;
}

@media (min-width: 1024px) {
  .emotion-46 {
    -webkit-flex: 1 0 auto;
    -ms-flex: 1 0 auto;
    flex: 1 0 auto;
    padding: 0 0.5rem;
    width: 33.3333%;
  }
}

.emotion-54 {
  -webkit-appearance: none;
  background: transparent;
  border-radius: 2px;
  color: currentColor;
  display: block;
  margin: 10px 0;
  width: 100%;
}

.emotion-54::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-54::-webkit-slider-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
  -webkit-appearance: none;
  margin-top: -9px;
}

.emotion-54:focus::-webkit-slider-runnable-track {
  background: currentColor;
}

.emotion-54::-moz-range-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-54::-moz-range-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-54::-ms-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: transparent;
  border-color: transparent;
  border-width: 8px 0;
  color: transparent;
}

.emotion-54::-ms-fill-lower {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-54::-ms-fill-upper {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-54::-ms-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-54:focus::-ms-fill-lower {
  background: currentColor;
}

.emotion-54:focus::-ms-fill-upper {
  background: currentColor;
}

.emotion-114 {
  padding: 5rem 0 1rem;
  text-align: center;
  width: 100%;
}

.emotion-118 {
  list-style: none;
  margin: 0 0 1rem;
  padding: 0;
}

.emotion-120 {
  display: inline-block;
  padding: 0 1rem;
}

.emotion-122 {
  color: currentColor;
  -webkit-transition: color 400ms ease-in;
  transition: color 400ms ease-in;
}

<div
  className="emotion-0 emotion-1"
>
  <main
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-4 emotion-5"
    >
      <div
        className="emotion-6 emotion-7"
      >
        <h1
          aria-atomic="true"
          aria-live="polite"
          className="emotion-8 emotion-9"
          data-testid="contrastResults-heading"
        >
          Yup
        </h1>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-12 emotion-13"
          >
            Small Text
            <br />
            AA: 4.5 AAA: 7.0
          </p>
          <h2
            className="emotion-14 emotion-15"
            data-testid="contrastResult-rating-small"
          >
            AA
          </h2>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-12 emotion-13"
          >
            <strong>
              Bold Text 18px and over 
              <br />
              AA: 3.0 AAA: 4.5
            </strong>
          </p>
          <h2
            className="emotion-14 emotion-15"
            data-testid="contrastResult-rating-bold"
          >
            AAA
          </h2>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-24 emotion-13"
          >
            Large Text 24px and over 
            <br />
            AA: 3.0 AAA: 4.5
          </p>
          <h2
            className="emotion-14 emotion-15"
            data-testid="contrastResult-rating-large"
          >
            AAA
          </h2>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-12 emotion-13"
          >
            Contrast Ratio
          </p>
          <h2
            className="emotion-14 emotion-15"
          >
            6.94 : 1
          </h2>
        </div>
      </div>
    </div>
    <div
      className="emotion-34 emotion-5"
    >
      <div
        className="emotion-36 emotion-37"
        data-testid="color-input-form"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <div
          className="emotion-38 emotion-39"
        >
          <label
            className="emotion-40 emotion-41"
            htmlFor="textColor"
          >
            Text Color
          </label>
          <input
            autoComplete="off"
            className="emotion-42 emotion-43"
            data-testid="textColor"
            id="textColor"
            onChange={[Function]}
            style={
              Object {
                "borderColor": "#fff",
                "color": "inherit",
              }
            }
            type="text"
            value="#eee"
          />
          <div
            className="emotion-44 emotion-45"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="textColor-hsl-Hue"
                >
                  Hue 0Â°
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="textColor-hsl-Hue"
                  id="textColor-hsl-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="textColor-hsl-Saturation"
                >
                  Saturation 0%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="textColor-hsl-Saturation"
                  id="textColor-hsl-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="textColor-hsl-Lightness"
                >
                  Lightness 93%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="textColor-hsl-Lightness"
                  id="textColor-hsl-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={93}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="emotion-38 emotion-39"
        >
          <label
            className="emotion-40 emotion-41"
            htmlFor="background"
          >
            Background
          </label>
          <input
            autoComplete="off"
            className="emotion-42 emotion-43"
            data-testid="background"
            id="background"
            onChange={[Function]}
            style={
              Object {
                "borderColor": "#fff",
                "color": "inherit",
              }
            }
            type="text"
            value="#5E38A8"
          />
          <div
            className="emotion-44 emotion-45"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="background-hsl-Hue"
                >
                  Hue 260Â°
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="background-hsl-Hue"
                  id="background-hsl-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={260}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="background-hsl-Saturation"
                >
                  Saturation 50%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="background-hsl-Saturation"
                  id="background-hsl-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={50}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="background-hsl-Lightness"
                >
                  Lightness 44%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="background-hsl-Lightness"
                  id="background-hsl-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={44}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="emotion-114 emotion-115"
  >
    <nav
      className="emotion-116 emotion-117"
    >
      <ul
        className="emotion-118 emotion-119"
      >
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Home
          </a>
        </li>
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/palette"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Palette
          </a>
        </li>
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/api-page"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            API
          </a>
        </li>
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            About
          </a>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;

exports[`Home Page should render without crashing 1`] = `
@media (min-width: 768px) {
  .emotion-0 {
    height: 100%;
  }
}

.emotion-2 {
  position: relative;
}

@media (min-width: 768px) {
  .emotion-2 {
    min-height: 85%;
  }
}

.emotion-4 {
  margin: 0 auto;
  max-width: 1150px;
  padding: 0 1rem;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 2rem;
  text-align: center;
}

@media (min-width: 768px) {
  .emotion-6 {
    margin-bottom: 3rem;
  }
}

@media (min-width: 1024px) {
  .emotion-6 {
    margin-bottom: 5rem;
  }
}

.emotion-8 {
  font-weight: 700;
  font-size: 20vw;
  margin: 3rem 0;
  width: 100%;
}

@media (min-width: 768px) {
  .emotion-8 {
    font-size: 18vw;
    margin: 4rem 0;
  }
}

@media (min-width: 1024px) {
  .emotion-8 {
    font-size: 15vw;
    margin: 5rem 0;
  }
}

@media (min-width: 1480px) {
  .emotion-8 {
    font-size: 9vw;
  }
}

.emotion-10 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-bottom: 2rem;
  width: 100%;
}

@media (min-width: 480px) {
  .emotion-10 {
    width: 50%;
  }
}

@media (min-width: 1024px) {
  .emotion-10 {
    width: 25%;
  }
}

.emotion-12 {
  font-size: 14pt;
  margin: 0;
}

.emotion-14 {
  font-weight: 700;
  font-size: 10vw;
  margin: 0;
}

@media (min-width: 768px) {
  .emotion-14 {
    font-size: 6vw;
  }
}

@media (min-width: 1024px) {
  .emotion-14 {
    font-size: 4vw;
  }
}

@media (min-width: 1480px) {
  .emotion-14 {
    font-size: 3vw;
  }
}

.emotion-24 {
  font-size: 18pt;
  margin: 0;
}

.emotion-34 {
  margin: 0 auto;
  max-width: 1200px;
}

.emotion-36 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 auto;
  max-width: 1200px;
  padding: 0 1rem;
  -webkit-transition: color 400ms ease-out;
  transition: color 400ms ease-out;
}

.emotion-38 {
  padding: 1rem;
  width: 100%;
}

@media (min-width: 768px) {
  .emotion-38 {
    -webkit-flex: 1 1 auto;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
    width: 50%;
  }
}

.emotion-40 {
  display: inline-block;
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-42 {
  background: rgba(255, 255, 255, 0.1);
  border: solid 1px;
  border-radius: 3px;
  font-size: 1.2rem;
  padding: 0.5rem 1rem;
  margin-bottom: 2rem;
  width: 100%;
}

.emotion-44 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (min-width: 1024px) {
  .emotion-44 {
    margin: 0 -0.5rem;
  }
}

.emotion-46 {
  width: 100%;
}

@media (min-width: 1024px) {
  .emotion-46 {
    -webkit-flex: 1 0 auto;
    -ms-flex: 1 0 auto;
    flex: 1 0 auto;
    padding: 0 0.5rem;
    width: 33.3333%;
  }
}

.emotion-54 {
  -webkit-appearance: none;
  background: transparent;
  border-radius: 2px;
  color: currentColor;
  display: block;
  margin: 10px 0;
  width: 100%;
}

.emotion-54::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-54::-webkit-slider-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
  -webkit-appearance: none;
  margin-top: -9px;
}

.emotion-54:focus::-webkit-slider-runnable-track {
  background: currentColor;
}

.emotion-54::-moz-range-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-54::-moz-range-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-54::-ms-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: transparent;
  border-color: transparent;
  border-width: 8px 0;
  color: transparent;
}

.emotion-54::-ms-fill-lower {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-54::-ms-fill-upper {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-54::-ms-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-54:focus::-ms-fill-lower {
  background: currentColor;
}

.emotion-54:focus::-ms-fill-upper {
  background: currentColor;
}

.emotion-114 {
  padding: 5rem 0 1rem;
  text-align: center;
  width: 100%;
}

.emotion-118 {
  list-style: none;
  margin: 0 0 1rem;
  padding: 0;
}

.emotion-120 {
  display: inline-block;
  padding: 0 1rem;
}

.emotion-122 {
  color: currentColor;
  -webkit-transition: color 400ms ease-in;
  transition: color 400ms ease-in;
}

<div
  className="emotion-0 emotion-1"
>
  <main
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-4 emotion-5"
    >
      <div
        className="emotion-6 emotion-7"
      >
        <h1
          aria-atomic="true"
          aria-live="polite"
          className="emotion-8 emotion-9"
          data-testid="contrastResults-heading"
        >
          Yup
        </h1>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-12 emotion-13"
          >
            Small Text
            <br />
            AA: 4.5 AAA: 7.0
          </p>
          <h2
            className="emotion-14 emotion-15"
            data-testid="contrastResult-rating-small"
          >
            AA
          </h2>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-12 emotion-13"
          >
            <strong>
              Bold Text 18px and over 
              <br />
              AA: 3.0 AAA: 4.5
            </strong>
          </p>
          <h2
            className="emotion-14 emotion-15"
            data-testid="contrastResult-rating-bold"
          >
            AAA
          </h2>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-24 emotion-13"
          >
            Large Text 24px and over 
            <br />
            AA: 3.0 AAA: 4.5
          </p>
          <h2
            className="emotion-14 emotion-15"
            data-testid="contrastResult-rating-large"
          >
            AAA
          </h2>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <p
            className="emotion-12 emotion-13"
          >
            Contrast Ratio
          </p>
          <h2
            className="emotion-14 emotion-15"
          >
            4.66 : 1
          </h2>
        </div>
      </div>
    </div>
    <div
      className="emotion-34 emotion-5"
    >
      <div
        className="emotion-36 emotion-37"
        data-testid="color-input-form"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <div
          className="emotion-38 emotion-39"
        >
          <label
            className="emotion-40 emotion-41"
            htmlFor="textColor"
          >
            Text Color
          </label>
          <input
            autoComplete="off"
            className="emotion-42 emotion-43"
            data-testid="textColor"
            id="textColor"
            onChange={[Function]}
            style={
              Object {
                "borderColor": "#fff",
                "color": "inherit",
              }
            }
            type="text"
            value="#FFFFFF"
          />
          <div
            className="emotion-44 emotion-45"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="textColor-hsl-Hue"
                >
                  Hue 0Â°
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="textColor-hsl-Hue"
                  id="textColor-hsl-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="textColor-hsl-Saturation"
                >
                  Saturation 0%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="textColor-hsl-Saturation"
                  id="textColor-hsl-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="textColor-hsl-Lightness"
                >
                  Lightness 100%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="textColor-hsl-Lightness"
                  id="textColor-hsl-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={100}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="emotion-38 emotion-39"
        >
          <label
            className="emotion-40 emotion-41"
            htmlFor="background"
          >
            Background
          </label>
          <input
            autoComplete="off"
            className="emotion-42 emotion-43"
            data-testid="background"
            id="background"
            onChange={[Function]}
            style={
              Object {
                "borderColor": "#fff",
                "color": "inherit",
              }
            }
            type="text"
            value="#1276CE"
          />
          <div
            className="emotion-44 emotion-45"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="background-hsl-Hue"
                >
                  Hue 208Â°
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="background-hsl-Hue"
                  id="background-hsl-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={208}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="background-hsl-Saturation"
                >
                  Saturation 84%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="background-hsl-Saturation"
                  id="background-hsl-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={84}
                />
              </div>
            </div>
            <div
              className="emotion-46 emotion-47"
            >
              <div
                className="emotion-48 emotion-49"
              >
                <label
                  className="emotion-40 emotion-41"
                  htmlFor="background-hsl-Lightness"
                >
                  Lightness 44%
                </label>
              </div>
              <div
                className="emotion-52 emotion-53"
              >
                <input
                  className="emotion-54 emotion-55"
                  data-testid="background-hsl-Lightness"
                  id="background-hsl-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={44}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="emotion-114 emotion-115"
  >
    <nav
      className="emotion-116 emotion-117"
    >
      <ul
        className="emotion-118 emotion-119"
      >
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Home
          </a>
        </li>
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/palette"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Palette
          </a>
        </li>
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/api-page"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            API
          </a>
        </li>
        <li
          className="emotion-120 emotion-121"
        >
          <a
            className="emotion-122 emotion-123"
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            About
          </a>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;
